File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESurfaceDimDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceDimDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd10
            AppTheme.STANDARD -> R.color.cstd10
            AppTheme.BLUE -> R.color.cbd10
            AppTheme.GREEN -> R.color.cgd10
            AppTheme.RED -> R.color.crd10
            AppTheme.NOTHING -> R.color.cnod10
        }
    }
}

File: lib/layer2_core/dark/e_surface_dim_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESurfaceDimDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd10;
      case 'standard': return cstd10;
      case 'blue': return cbd10;
      case 'green': return cgd10;
      case 'red': return crd10;
      case 'nothing': return cnod10;
      default: return cstd10;
    }
  }
}
